bpf/cpu/cpu.bpf.c

```diff
@@ -935,37 +965,41 @@ int profile_cpu(struct bpf_perf_event_data *ctx) {
     }
   }
 
-  bool has_unwind_info = has_unwind_information(user_pid);
-  // Check if the process is eligible for the unwind table or frame pointer
-  // unwinders.
-  if (!has_unwind_info) {
-    add_stack(ctx, pid_tgid, STACK_WALKING_METHOD_FP, NULL);
-  } else {
-    set_initial_state(&ctx->regs);
-
-    u32 zero = 0;
-    unwind_state_t *unwind_state = bpf_map_lookup_elem(&heap, &zero);
-    if (unwind_state == NULL) {
-      // This should never happen.
-      return 0;
-    }
+  set_initial_state(&ctx->regs);
+  u32 zero = 0;
+  unwind_state_t *unwind_state = bpf_map_lookup_elem(&heap, &zero);
+  if (unwind_state == NULL) {
+    // This should never happen.
+    return 0;
+  }
 
+  if (has_unwind_information(user_pid)) {
     shard_info_t *shard = NULL;
     find_unwind_table(&shard, user_pid, unwind_state->ip, NULL);
     if (shard == NULL) {
-      bpf_printk("IP not covered. In kernel space / JIT / bug? IP %llx)", unwind_state->ip);
```

Comment:
This is no longer valid because we are retrieving now the user registers saved when switching to kernel code during a syscall execution